sw/device/silicon_creator/lib/sigverify.c

```diff
@@ -201,26 +201,29 @@ static rom_error_t sigverify_encoded_message_check(
  * value.
  *
  * @param lc_state Life cycle state of the device.
- * @param[out] use_sw Use software implementation for signature verification.
- * @return Result of the operation.
+ * @return Whether to use software implementation for signature verification.
  */
-static rom_error_t sigverify_use_sw_rsa_verify(lifecycle_state_t lc_state,
-                                               hardened_bool_t *use_sw) {
-  switch (lc_state) {
+static hardened_bool_t sigverify_use_sw_rsa_verify(lifecycle_state_t lc_state) {
+  switch (launder32(lc_state)) {
```

Comment:
How about doing this instead? Assuming is difficult for the attacker to trigger `kLcStateTest`.

```cc
if (launder32(lc_state) == kLcStateTest) {
  HARDENED_CHECK_EQ(lc_state, kLcStateTest);
  return kHardenedBoolTrue;
}
return otp_read32(OTP_CTRL_PARAM_CREATOR_SW_CFG_USE_SW_RSA_VERIFY_OFFSET);
```

Reply:
Discussed this offline, will keep the switch statement as a redundant check for the life cycle state.